### PR TITLE
Mavlink - Configure Tx modules to work as standalone transmitters without a handset

### PR DESCRIPTION
**Pull request by JyeSmith**
This PR does a couple of things that are all targeted at Tx modules being used without a handset.

1. When no handset is detected (e.g. a Tx module powered via the XT30 or USB) the USB port is monitored for incoming data.  The data is check to see if it contains a mav packet.  If a mav packet is detected the OTA Linkmode is changed from Normal to MAVLink, and OTA transmission is started.  This means a completely default and non configured Tx module can be connected to Mission Planner, CONNECT pressed, and the Tx module will start to transmit in MAVLink mode.
2.  If the Backpack is flashed with mavesp the Tx module will start to transmit OTA without a handset.  Note that MAVLink mode must first be configured, either via Lua or by an initial USB connection as described in point 1.  The detection of a mav packet from the BP does not change the Linkmode.  This is so that a users can still switch the Normal via Lua and not have the BP override this change.
3. When no channel input is detected, no channel data is transmitted OTA.  We do not want to send fake data to the FC when there is no handset connected.  In place of the RC data empty MSP packets are sent.  We must send MSP packets and not Sync packets for the required SS Confirmation bit.

Demo setup video https://youtu.be/EYOOlW20QDo?si=LHpaXRBSCUBlGtkW

### CODE REVIEW

File: src/lib/MAVLink/MAVLink.cpp [hunk 1]
```diff
@@ -105,3 +105,24 @@ void convert_mavlink_to_crsf_telem(uint8_t *CRSFinBuffer, uint8_t count, Handset
...
+#if !defined(PLATFORM_STM32)
+    for (uint8_t i = 0; i < bufferSize; ++i)
+    {
+        uint8_t c = buffer[i];
+
+        mavlink_message_t msg;
+        mavlink_status_t status;
+
+        // Try parse a mavlink message
+        if (mavlink_parse_char(MAVLINK_COMM_0, c, &msg, &status))
```
Comment (IamPete1) [inline src/lib/MAVLink/MAVLink.cpp:120]: I think you need to use another set of buffers here. `convert_mavlink_to_crsf_telem` is using `MAVLINK_COMM_0` 
```suggestion
        if (mavlink_parse_char(MAVLINK_COMM_1, c, &msg, &status))
```
Will need to bump `MAVLINK_COMM_NUM_BUFFERS` once https://github.com/ExpressLRS/ExpressLRS/pull/2846 is merged.

Comment (JyeSmith) [reply src/lib/MAVLink/MAVLink.cpp:120]: Will do and thanks for the review!

**Comment by commanderguy3001** [comment]
will this work with sending CRSF messages via usb uart as well? there's numerous people who have been messing around with sending control data from a PC, often with something like a gamepad as input.
it'd be very neat if that was seamless as well, without having to manually edit the hardware.html for that to be possible.

**Comment by JyeSmith** [comment]
> will this work with sending CRSF messages via usb uart as well? there's numerous people who have been messing around with sending control data from a PC, often with something like a gamepad as input. it'd be very neat if that was seamless as well, without having to manually edit the hardware.html for that to be possible.

No because it would require monitoring 2 UARTs, SPORT and the USB.

File: src/lib/MAVLink/MAVLink.cpp (hunk 1 above)
Comment (JyeSmith) [reply src/lib/MAVLink/MAVLink.cpp:120]: I was about to make this change but it mightnt be required.

This buffer is only used before the tx module starts to transmit and before a vehicle connection.  After a connection is made it is no longer used, and can be used freely by convert_mavlink_to_crsf_telem().

Is this ok, or is there something Im missing with how the buffer is being use?

**Comment by JyeSmith** [comment]
Updated to use mavlink_frame_char() as proposed in  https://github.com/ExpressLRS/ExpressLRS/pull/2858

**Review by MUSTARDTIGERFPV — APPROVED** [review]

**Review by pkendall64 — APPROVED** [review]

File: src/lib/MAVLink/MAVLink.cpp (hunk 1 above)
Comment (IamPete1) [reply src/lib/MAVLink/MAVLink.cpp]: Hum, if there never used at that same time then I guess its OK. If they do ever get used at once then messages will be interleaved resulting in corruption of both. Might be worth a comment to warn future us about that.

**Pull request merged**